Shared/Samples/Generate offline map with local basemap/GenerateOfflineMapWithLocalBasemapView.swift

```diff
@@ -0,0 +1,294 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct GenerateOfflineMapWithLocalBasemapView: View {
+    /// A Boolean value indicating whether the job is generating an offline map.
+    @State private var isGeneratingOfflineMap = false
+    
+    /// A Boolean value indicating when the basemap choice dialog is presented.
+    @State private var basemapChoiceIsPresented = false
+    
+    /// A Boolean value indicating whether the job is cancelling.
+    @State private var isCancellingJob = false
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    /// The title to show in the confirmation dialog.
+    let basemapChoiceTitle: String = {
+        let displayName = Bundle.main.object(forInfoDictionaryKey: "CFBundleName") as! String
+        return String(format: "“%@” has a local imagery basemap that can be used in the map. Would you like to use the local or the online basemap?", displayName)
+    }()
+    
+    /// The view model for this sample.
+    @StateObject private var model = Model()
+    
+    var body: some View {
+        GeometryReader { geometry in
+            MapViewReader { mapView in
+                MapView(map: model.offlineMap ?? model.onlineMap)
+                    .interactionModes(isGeneratingOfflineMap ? [] : [.pan, .zoom])
+                    .errorAlert(presentingError: $error)
+                    .task {
+                        do {
+                            try await model.initializeOfflineMapTask()
+                        } catch {
+                            self.error = error
+                        }
+                    }
+                    .onDisappear {
+                        Task { await model.cancelJob() }
+                    }
+                    .overlay {
+                        Rectangle()
+                            .stroke(.red, lineWidth: 2)
+                            .padding(EdgeInsets(top: 20, leading: 20, bottom: 44, trailing: 20))
+                            .opacity(model.offlineMap == nil ? 1 : 0)
+                    }
+                    .overlay {
+                        if isGeneratingOfflineMap,
+                           let progress = model.generateOfflineMapJob?.progress {
+                            VStack(spacing: 16) {
+                                ProgressView(progress)
+                                    .progressViewStyle(.linear)
+                                    .frame(maxWidth: 200)
+                                
+                                Button("Cancel") {
+                                    isCancellingJob = true
+                                }
+                                .disabled(isCancellingJob)
+                                .task(id: isCancellingJob) {
+                                    // Ensures cancelling the job is true.
```

Comment:
```suggestion
```

I'm not sure this comment is helpful. If you wanted to keep it, you could maybe say something like, "Ensures the job is being cancelled.", but I would personally omit it. Up to you.